plugin/plugin.go

```diff
@@ -23,35 +23,75 @@ const (
 
 	// CommandDescribeKey is the name of the plugin command
 	// which must be supported by every plugin that has the
-	// SIGNATURE_GENERATOR capability.
+	// SIGNATURE_GENERATOR.RAW capability.
 	CommandDescribeKey Command = "describe-key"
 
 	// CommandGenerateSignature is the name of the plugin command
 	// which must be supported by every plugin that has the
-	// SIGNATURE_GENERATOR capability.
+	// SIGNATURE_GENERATOR.RAW capability.
 	CommandGenerateSignature Command = "generate-signature"
 
 	// CommandGenerateEnvelope is the name of the plugin command
 	// which must be supported by every plugin that has the
-	// SIGNATURE_ENVELOPE_GENERATOR capability.
+	// SIGNATURE_GENERATOR.ENVELOPE capability.
 	CommandGenerateEnvelope Command = "generate-envelope"
+
+	// CommandVerifySignature is the name of the plugin command
+	// which must be supported by every plugin that has
+	// any SIGNATURE_VERIFIER.* capability
+	CommandVerifySignature Command = "verify-signature"
 )
 
 // Capability is a feature available in the plugin contract.
 type Capability string
 
 const (
-	// CapabilitySignatureGenerator is the name of the capability
-	// which should support a plugin to support generating signatures.
-	CapabilitySignatureGenerator Capability = "SIGNATURE_GENERATOR"
+	// CapabilityRawGenerator is the name of the capability
+	// for a plugin to support generating raw signatures.
+	CapabilityRawGenerator Capability = "SIGNATURE_GENERATOR.RAW"
 
 	// CapabilityEnvelopeGenerator is the name of the capability
-	// which should support a plugin to support generating envelope signatures.
-	CapabilityEnvelopeGenerator Capability = "SIGNATURE_ENVELOPE_GENERATOR"
+	// for a plugin to support generating envelope signatures.
+	CapabilityEnvelopeGenerator Capability = "SIGNATURE_GENERATOR.ENVELOPE"
+
+	// CapabilityTrustedIdentityVerifier is the name of the capability
+	// for a plugin to support verifying trusted identities.
+	CapabilityTrustedIdentityVerifier = Capability(VerificationCapabilityTrustedIdentity)
+
+	// CapabilityRevocationCheckVerifier is the name of the capability
+	// for a plugin to support verifying revocation checks.
+	CapabilityRevocationCheckVerifier = Capability(VerificationCapabilityRevocationCheck)
+)
+
+// VerificationCapability is a verification feature available in the plugin contract.
+type VerificationCapability string
```

Comment:
Is it necessary to create a new type for verification capabilities? Could we just add the new ones to `Capability`?

Reply:
`SIGNATURE_VERIFIER.TRUSTED_IDENTITY` and `SIGNATURE_VERIFIER.REVOCATION_CHECK` are the only two valid verification capabilities in the verify-signature request. It's cleaner to only allow the programmer to use valid values rather than include the other capabilities which do not apply.